bmcd/src/legacy.rs

```diff
@@ -0,0 +1,395 @@
+//! Routes for legacy API present in versions <= 1.1.0 of the firmware.
+
+use std::fmt::Display;
+use std::str::FromStr;
+use std::sync::Arc;
+
+use actix_web::{rt, web, HttpResponse};
+use nix::sys::statfs::statfs;
+use serde_json::json;
+use tokio::sync::{mpsc, Mutex};
+use tpi_rs::app::bmc_application::{BmcApplication, UsbConfig};
+use tpi_rs::middleware::{NodeId, UsbMode, UsbRoute};
+
+type Query = web::Query<std::collections::HashMap<String, String>>;
+
+trait ToHttpResponse {
+    fn to_response(&self, action: &'static str) -> HttpResponse;
+}
+
+impl<T, E: Display> ToHttpResponse for Result<T, E> {
+    fn to_response(&self, action: &'static str) -> HttpResponse {
+        match self {
+            Ok(_) => success_response(),
+            Err(e) => {
+                let msg = format!("Failed to {}: {}", action, e);
+                HttpResponse::InternalServerError().body(msg)
+            }
+        }
+    }
+}
+
+fn success_response() -> HttpResponse {
+    // For backwards compatibility, `response` holds an array, even though it's unnecessary
+    let msg = json!({
+        "response": [{ "result": "ok" }]
+    });
+
+    HttpResponse::Ok().json(msg)
+}
+
+pub fn config(cfg: &mut web::ServiceConfig) {
+    cfg.service(
+        web::scope("/api/bmc").service(
+            web::resource("")
+                .route(web::get().to(api_entry))
+                .route(web::post().to(api_post)),
+        ),
+    );
+}
+
+// TODO: BmcApplication::new() needs to return just BmcApplication, not wrapped in Arc<>.
+async fn api_entry(bmc: web::Data<Mutex<Arc<BmcApplication>>>, query: Query) -> HttpResponse {
+    let is_set = match query.get("opt").cloned().as_deref() {
```

Comment:
nitpick: you can avoid the clone by doing:
 
```suggestion
    let is_set = match query.get("opt").map(String::as_str) {
```

Reply:
Done